src/webgpu/shader/execution/builtin/abs.spec.ts

```diff
@@ -0,0 +1,301 @@
+        { input: -8192.0, expected: [8192.0] },
+        { input: -16384.0, expected: [16384.0] },
+        { input: -32768.0, expected: [32768.0] },
+        { input: -65536.0, expected: [65536.0] },
+        { input: -131072.0, expected: [131072.0] },
+        { input: -262144.0, expected: [262144.0] },
+        { input: -524288.0, expected: [524288.0] },
+        { input: -1048576.0, expected: [1048576.0] },
+        { input: -8388607.0, expected: [8388607.0] }, //2^23 - 1
+        { input: -16777215.0, expected: [16777215.0] }, //2^24 - 1
+        { input: -16777216.0, expected: [16777216.0] }, //2^24
+      ]
+    );
+  });
+
+g.test('abs_int')
+  .uniqueId(0xfab878e682c16d42)
+  .desc(
+    `
+https://www.w3.org/TR/2021/WD-WGSL-20210831#integer-builtin-functions
+signed abs:
+T is i32 or vecN<i32> abs(e: T ) -> T The absolute value of e.
+Component-wise when T is a vector.
+If e evaluates to the largest negative value, then the result is e.
+`
+  )
+  .params(u =>
+    u
+      .combineWithParams([
+        { storageClass: 'storage', storageMode: 'read_write', access: 'read' },
+      ] as const)
+      .combineWithParams([{ containerType: 'scalar' }, { containerType: 'vector' }] as const)
+      .combineWithParams([{ isAtomic: false }] as const)
+      .combineWithParams([{ baseType: 'i32' }] as const)
+      .beginSubcases()
+      .expandWithParams(generateTypes)
+  )
+  .fn(async t => {
+    assert(t.params._kTypeInfo !== undefined, 'generated type is undefined');
+    runShaderTest(
+      t,
+      t.params.storageClass,
+      t.params.storageMode,
+      t.params.baseType,
+      t.params.type,
+      t.params._kTypeInfo.arrayLength,
+      'abs',
+      OperandType.Int,
+      [
+        { input: -1, expected: [1] },
+        { input: -2, expected: [2] },
+        { input: -4, expected: [4] },
+        { input: -8, expected: [8] },
+        { input: -16, expected: [16] },
+        { input: -32, expected: [32] },
+        { input: -64, expected: [64] },
+        { input: -128, expected: [128] },
+        { input: -256, expected: [256] },
+        { input: -512, expected: [512] },
+        { input: -1024, expected: [1024] },
+        { input: -2048, expected: [2048] },
+        { input: -4096, expected: [4096] },
+        { input: -8192, expected: [8192] },
+        { input: -16384, expected: [16384] },
+        { input: -32768, expected: [32768] },
+        { input: -65536, expected: [65536] },
+        { input: -131072, expected: [131072] },
+        { input: -262144, expected: [262144] },
+        { input: -524288, expected: [524288] },
+        { input: -1048576, expected: [1048576] },
+        { input: -8388607, expected: [8388607] },
+        { input: -16777215, expected: [16777215] },
+        { input: -16777216, expected: [16777216] },
+        { input: -134217727, expected: [134217727] },
+        { input: -1073741823, expected: [1073741823] },
+        { input: -2147483647, expected: [2147483647] },
```

Comment:
Does `abs(-2147483648)` have a defined value that we can test? (I would guess it's `-2147483648`?)

Reply:
defined value? you mean define a constant and use that instead of the gibberish number? (if yes, then my next pr fixes it).

Reply:
I just mean does WGSL's definition of abs() have a well-specified result for `abs(-0x8000_0000)`.
And in fact it does: https://gpuweb.github.io/gpuweb/wgsl/#integer-builtin-functions
> If e evaluates to the largest negative value, then the result is e.

Reply:
So we should test that corner case.